tools/wpt/revlist.py

```diff
@@ -0,0 +1,205 @@
+    # type: (**Any) -> Number]
+    num_revisions = kwargs.get("num_revisions", DEFAULT_NUM_REVISIONS)
+    return int(num_revisions)
+
+
+def parser_verbose(**kwargs):
+    # type: (**Any) -> Boolean]
+    return kwargs.get("verbose", DEFAULT_VERBOSE)
+
+
+COMMAND_ARGS = {
+    "epoch_until": {
+        "name": "epoch-until",
+        "default": DEFAULT_EPOCH_UNTIL,
+        "help": "Show revisions older than a specific date in UNIX "
+                "timestamp. NOW is the default value. For example: "
+                "'1570345200'.",
+        "parser": parser_epoch_until
+    },
+    "epoch_step": {
+        "name": "epoch-step",
+        "default": DEFAULT_EPOCH_STEP,
+        "help": "Regular interval of seconds used to get the tagged revision. "
+                "hourly, two-hourly, four-hourly, six-hourly, eight-hourly, "
+                "twelve-hourly, daily, weekly or 3600, 7200, 86000 ...",
+        "parser": parser_epoch_step
+    },
+    "num_revisions": {
+        "name": "num-revisions",
+        "default": DEFAULT_NUM_REVISIONS,
+        "help": "Number of revisions to get.",
+        "parser": parser_num_revisions
+    },
+    "log_level": {
+        "name": "log-level",
+        "default": DEFAULT_LOG_LEVEL,
+        "help": "Debug log level. For example: DEBUG, INFO, WARNING, ERROR, "
+                "CRITICAL or an integer value like 10, 20 ",
+        "parser": parser_debug_log_level
+    },
+    "verbose": {
+        "name": "verbose",
+        "default": DEFAULT_VERBOSE,
+        "help": "Show more information in the output",
+        "parser": parser_verbose,
+        "type": bool
+    }
+}
+
+
+def get_git_cmd(repo_path):
+    # type: (bytes) -> Callable[..., Text]
+    """Create a function for invoking git commands as a subprocess."""
+    def git(cmd, *args):
+        # type: (Text, *Union[bytes, Text]) -> Text
+        full_cmd = [u"git", cmd] + list(item.decode("utf8") if isinstance(item, bytes) else item for item in args)  # type: List[Text]
+        try:
+            logger.debug(" ".join(full_cmd))
+            return subprocess.check_output(full_cmd, cwd=repo_path).decode("utf8").strip()
+        except subprocess.CalledProcessError as e:
+            logger.critical("Git command exited with status %i" % e.returncode)
+            logger.critical(e.output)
+            sys.exit(1)
+    return git
+
+
+def list_tagged_revisons(**kwargs):
+    # type: (**Any) -> List[Text]
+    logger.debug("list_tagged_revisons: %s" % kwargs)
+    git = get_git_cmd(wpt_root)
+    epoch_step = COMMAND_ARGS["epoch_step"]["parser"](**kwargs)
+    epoch_until = COMMAND_ARGS["epoch_until"]["parser"](**kwargs)
+    num_revisions = COMMAND_ARGS["num_revisions"]["parser"](**kwargs)
+    verbose = COMMAND_ARGS["verbose"]["parser"](**kwargs)
+
+    iterator = 0
+    result = []
+    previous_command_output = ""
+    while iterator < num_revisions:
+        min_age = (( epoch_until / epoch_step ) - iterator) * epoch_step
```

Comment:
Because `epoch_until` is defined in terms of the current time and the script can take a while to run, I suspect this doesn't produce entirely predictable output if the script takes a while to run.

This will probably go away as a side effect of avoiding one `git rev-list` call per iteration, so I'll just leave this comment here to follow up on later if that isn't the case.

Reply:
`epoch_until` is going to be the same for all the iterations. Also  `min_age` is defined basically as a kind of *floor* value  of  `epoch_until`  and always older or equal than the `epoch_until` value. The possible new commits should be always filter out by `min_age`. 

The only case that it could be problematic could be the situation described in https://github.com/web-platform-tests/wpt/issues/19322#issuecomment-538996990

Reply:
Resolved by `calculate_min_timestamp` clamping to an epoch boundary.